### PR TITLE
COZMO-9128 Camera properties and exposure controls

### PR DESCRIPTION
**Pull request by MarkWesleyAtAnki**
Added a Vector2 helper class (like Vector3, but without the z component)
Added a CameraConfig class to robot for keeping track of camera properties and settings
On connection SDK programs now immediately receive a number of per-robot settings, includes serial number and camera calibration.
Camera Exposure can now be override manually.
Current exposure and gain values are streamed back to SDK from Engine as they’re updated.
Added a tutorial example that shows exposure settings in action
Expanded annotation system slightly to make it support outlined/shadowed text, and line-spacing overrides.

### CODE REVIEW

File: src/cozmo/annotate.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2016 Anki, Inc.
+# Copyright (c) 2016-2017 Anki, Inc.
```
Comment (msintov) [inline src/cozmo/annotate.py]: All SDK and example files should be updated to 2017, perhaps in a separate PR?

File: examples/tutorials/03_vision/04_exposure.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+try:
+    from PIL import ImageDraw, ImageFont
+    import numpy as np
+except ImportError:
+    sys.exit('run `pip3 install --user Pillow numpy` to run this example')
+
+import cozmo
+
+
+# A global string value to display in the window to make it more obvious
```
Comment (msintov) [inline examples/tutorials/03_vision/04_exposure.py]: Might be nice to specify what window is implied (i.e., camera viewer window).

Also should specify that the program will update this global. Initially I was thinking that the user would add a setting on this line.

File: examples/tutorials/03_vision/04_exposure.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+
+# An annotator for live-display of all of the camera info
+@cozmo.annotate.annotator
+def camera_info(image, scale, annotator=None, world=None, **kw):
+    d = ImageDraw.Draw(image)
+    bounds = [3, 0, image.width, image.height]
+
+    camera_config = world.robot.camera_config
+    text_to_display = "Example Mode: " + example_mode + "\n\n"
```
Comment (msintov) [inline examples/tutorials/03_vision/04_exposure.py]: Could add a comment specifying where this text_to_display will display (i.e., in the viewer window).

File: examples/tutorials/03_vision/04_exposure.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                                    color="white",
+                                    outline_color="black", full_outline=True)
+    text.render(d, bounds)
+
+
+def cozmo_program(robot: cozmo.robot.Robot):
+    global example_mode
+    robot.world.image_annotator.add_annotator('camera_info', camera_info)
+
+    # Ensure camera is in auto-exposure, demo for 5 seconds
```
Comment (msintov) [inline examples/tutorials/03_vision/04_exposure.py]: I was confused by "demo for 5 seconds." Maybe  make that a second sentence that says something like, "Demonstrate auto-exposure for 5 seconds."?

File: examples/tutorials/03_vision/04_exposure.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    robot.enable_auto_exposure()
+    example_mode = "Auto Exposure"
+    time.sleep(5)
+
+    # Grab the current auto exposure/gain values as good defaults for
```
Comment (msintov) [inline examples/tutorials/03_vision/04_exposure.py]: It looks like sometimes auto-exposure is hyphenated and sometimes not?

File: examples/tutorials/03_vision/04_exposure.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    # the current lighting conditions
+    cam = robot.camera_config
+    auto_exposure_ms = cam.exposure_ms
+    auto_gain = cam.camera_gain
+
+    # Demonstrate Manual Exposure, linearly increasing the exposure time
```
Comment (msintov) [inline examples/tutorials/03_vision/04_exposure.py]: Could leave "Manual Exposure" as all lowercase

File: examples/tutorials/03_vision/04_exposure.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    example_mode = "Manual Exposure - Increasing Exposure, Fixed Gain"
+    for exposure in range(cam.min_exposure_time_ms, cam.max_exposure_time_ms+1, 1):
+        robot.set_manual_exposure(exposure, auto_gain)
+        time.sleep(0.1)
+
+    # Demonstrate Manual Exposure, linearly increasing the gain
```
Comment (msintov) [inline examples/tutorials/03_vision/04_exposure.py]: Same as above

File: src/cozmo/robot.py [hunk 8]
```diff
@@ -803,6 +930,19 @@ def has_in_progress_actions(self):
...
         return self._action_dispatcher.has_in_progress_actions
 
+    @property
+    def camera_config(self):
+        ''':class:`cozmo.robot.CameraConfig`: The read-only config/calibration for this robot's camera'''
+        return self._camera_config
+
+    @property
+    def serial(self):
+        '''string: The serial number, as a hex-string, for the robot.
```
Comment (msintov) [inline src/cozmo/robot.py]: Might be nice to provide an example of what this value might look like.

File: src/cozmo/annotate.py (hunk 1 above)
Comment (MarkWesleyAtAnki) [inline src/cozmo/annotate.py]: Yeah, I believe new files should be 2017 now, and as we modify 2016 files they should become 2016-2017. Will clean any stragglers up in a later PR, I'm led to believe this isn't that critical given copyright duration.

File: examples/tutorials/03_vision/04_exposure.py (hunk 5 above)
Comment (MarkWesleyAtAnki) [inline examples/tutorials/03_vision/04_exposure.py]: I've switched to non hyphenated

File: examples/tutorials/03_vision/04_exposure.py (hunk 3 above)
Comment (MarkWesleyAtAnki) [inline examples/tutorials/03_vision/04_exposure.py]: I've expanded the comment above the annotator a couple of lines up to clarify.

**Pull request merged**